src/Data/Repositories/FUTXORepository.cs

```diff
@@ -50,6 +53,38 @@ public async Task<List<FMUTXO>> GetAll()
             throw new NotImplementedException();
         }
 
+        public async Task<List<FMUTXO>> GetByWalletId(int walletId)
+        {
+            await using var applicationDbContext = await _dbContextFactory.CreateDbContextAsync();
+            
+            return await applicationDbContext.FMUTXOs.Where(x => x.Wallet.Id == walletId)
+                .Include(x => x.Tags).ToListAsync();
+        }
+
+        public async Task<List<FMUTXO>> GetFromUTXOs(List<FMUTXO> utxos)
+        {
+            await using var applicationDbContext = await _dbContextFactory.CreateDbContextAsync();
+            
+            var allUtxos = await applicationDbContext.FMUTXOs
+                .ToListAsync();
+
+            List<FMUTXO> newUtxos = new List<FMUTXO>();
+            foreach (var utxo in allUtxos)
+            {
+                if (utxos.Contains(utxo))
+                {
+                    newUtxos.Add(utxo);
+                }
+            }
```

Comment:
I've used this `foreach` because the linq call was not working for checking if the utxo checked was on the list. That `contains` call inside the linq call was not using our `equals`.